larrecodnn/ImagePatternAlgs/Tensorflow/Modules/WaveformRoiFinder_module.cc

```diff
@@ -76,8 +80,12 @@ nnet::WaveformRoiFinder::WaveformRoiFinder(fhicl::ParameterSet const& p)
   }
 
   // Signal/Noise waveform recognition tool
-  fWaveformRecogTool =
-    art::make_tool<wavrec_tool::IWaveformRecog>(p.get<fhicl::ParameterSet>("WaveformRecog"));
+  fWaveformRecogToolVec.resize(fNPlanes);
+  for (int n=0; n<fNPlanes; n++) {
+    char planeWaveformRecog[30];
+    sprintf(planeWaveformRecog, "WaveformRecogPlane%d",n);
+    fWaveformRecogToolVec[n] = art::make_tool<wavrec_tool::IWaveformRecog>(p.get<fhicl::ParameterSet>(planeWaveformRecog));
+  }
```

Comment:
Assuming you really need each plane to have different tools, I suggest a different approach--the configuration file should specify a sequence of configurations for each tool (see FHiCL file comment), and then the tools can be easily constructed using:
```.cpp
fWaveformRecogToolVec.reserve(fNPlanes);
auto const tool_psets = p.get<std::vector<fhicl::ParameterSet>>("WaveformRecogs");
for (auto const& pset : tool_psets) {
  fWaveformRecogToolVec.push_back(art::make_tool<wavrec_tool::IWaveformRecog>(pset));
}
```